node/src/command-options/LPosOptions.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+export class LPosOptions implements LPosOptions {
+    public static RANK_REDIS_API = "RANK";
+    public static COUNT_REDIS_API = "COUNT";
+    public static MAXLEN_REDIS_API = "MAXLEN";
+    private rank?: number;
+    private count?: number;
+    private maxLength?: number;
+
+    public setRank(rank: number): void {
+        this.rank = rank;
+    }
+
+    public setCount(count: number): void {
+        this.count = count;
+    }
+
+    public setMaxLength(maxLength: number): void {
+        this.maxLength = maxLength;
+    }
+
+    public toArgs(): string[] {
+        const args: string[] = [];
+
+        if (this.rank !== undefined) {
+            args.push(LPosOptions.RANK_REDIS_API);
+            args.push(this.rank.toString());
+        }
+
+        if (this.count !== undefined) {
+            args.push(LPosOptions.COUNT_REDIS_API);
+            args.push(this.count.toString());
+        }
+
+        if (this.maxLength !== undefined) {
+            args.push(LPosOptions.MAXLEN_REDIS_API);
+            args.push(this.maxLength.toString());
+        }
+
+        return args;
+    }
+}
+
+export class LPosOptionsBuilder {
```

Comment:
I don't think that we need a builder. It is a mostly java thing to bypass lang limitations (no named args, etc).
```ts
const options: LPosOptions = {
   rank: 1,
   maxLength: 2
}
```
and then
```ts
const res = await client.lpos(key, elem, options);
```
or something like this
```ts
const res = await client.lpos(key, elem, new LPosOptions({ rank: 1, maxLength: 2 }));
const res = await client.lpos(key, elem, { rank: 1, maxLength: 2 });
```
ref https://stackoverflow.com/a/45514481/21176342